### PR TITLE
Enable/Disable YANG ConfigDB update in minigraph

### PR DESCRIPTION
**Pull request by isabelmsft**
#### Why I did it
Disable YANG validation of updates to Config DB by default in minigraph
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: Makefile.work [hunk 1]
```diff
@@ -43,6 +43,7 @@
 #  * ENABLE_AUTO_TECH_SUPPORT: Enable the configuration for event-driven techsupport & coredump mgmt feature
 #  *                 Default: y
 #  *                 Values: y,n
+#  * ENABLE_YANG_CONFIG_VALIDATION: Enable YANG validation of updates to ConfigDB
```
Comment (qiluo-msft) [inline Makefile.work:46]: >ENABLE_YANG_CONFIG_VALIDATION [](http://example.com/codeflow?start=5&length=29)

This name is too general, considering we already use yang model to validate some config.

Suggest `ENABLE_CLI_YANG_VALIDATION`

File: slave.mk [hunk 2]
```diff
@@ -842,7 +844,10 @@ $(addprefix $(TARGET_PATH)/, $(DOCKER_IMAGES)) : $(TARGET_PATH)/%.gz : .platform
 		$(eval export $(subst -,_,$(notdir $($*.gz_PATH)))_dbgs=$(shell printf "$(subst $(SPACE),\n,$(call expand,$($*.gz_DBG_PACKAGES)))\n" | awk '!a[$$0]++'))
 		$(eval export $(subst -,_,$(notdir $($*.gz_PATH)))_pkgs=$(shell printf "$(subst $(SPACE),\n,$(call expand,$($*.gz_APT_PACKAGES)))\n" | awk '!a[$$0]++'))
 		if [ -d $($*.gz_PATH)/cli-plugin-tests/ ]; then pushd $($*.gz_PATH)/cli-plugin-tests; pytest-$($(SONIC_UTILITIES_PY3)_PYTHON_VERSION) -v $(LOG); popd; fi
-		# Label docker image with componenets versions
+		if [ -f $($*.gz_PATH)/init_cfg.json.j2 ] ; then
+			j2 $($*.gz_PATH)/init_cfg.json.j2 > $($*.gz_PATH)/init_cfg.json
```
Comment (qiluo-msft) [inline slave.mk:848]: >j2 $($*.gz_PATH)/init_cfg.json.j2 > $($*.gz_PATH)/init_cfg.json [](http://example.com/codeflow?start=3&length=63)

Is this line really needed?

File: Makefile.work (hunk 1 above)
Comment (qiluo-msft) [reply Makefile.work:46]: I realize the `CLI` is not a proper name either. Let's keep this discussion open for a while.

File: files/build_templates/init_cfg.json.j2 [hunk 3]
```diff
@@ -4,7 +4,8 @@
             "buffer_model": {% if default_buffer_model == "dynamic" %}"dynamic"{% else %}"traditional"{% endif %},
             {%- if include_p4rt == "y" %}"synchronous_mode":"enable",{% endif %}
             "default_bgp_status": {% if shutdown_bgp_on_start == "y" %}"down"{% else %}"up"{% endif %},
-            "default_pfcwd_status": {% if enable_pfcwd_on_start == "y" %}"enable"{% else %}"disable"{% endif %}
+            "default_pfcwd_status": {% if enable_pfcwd_on_start == "y" %}"enable"{% else %}"disable"{% endif %},
+            "yang_config_validation": {% if enable_yang_config_validation == "y" %}"enable"{% else %}"disable"{% endif %}
```
Comment (qiluo-msft) [inline files/build_templates/init_cfg.json.j2:8]: >yang_config_validation [](http://example.com/codeflow?start=13&length=22)

All other parameters are "default_*", is your intention is the same, like "default_enable_..."?

File: platform/vs/docker-sonic-vs/Dockerfile.j2 [hunk 4]
```diff
@@ -168,6 +168,7 @@ COPY ["files/configdb-load.sh", "/usr/bin/"]
 COPY ["files/arp_update", "/usr/bin/"]
 COPY ["files/buffers_config.j2", "files/qos_config.j2", "files/arp_update_vars.j2", "files/copp_cfg.j2", "/usr/share/sonic/templates/"]
 COPY ["files/sonic_version.yml", "/etc/sonic/"]
+COPY ["init_cfg.json", "/etc/sonic/"]
```
Comment (qiluo-msft) [inline platform/vs/docker-sonic-vs/Dockerfile.j2:171]: Is it conflicting with below `COPY init_cfg.json.j2`?

File: platform/vs/docker-sonic-vs/start.sh [hunk 5]
```diff
@@ -33,8 +33,13 @@ mkdir -p /var/run/redis/sonic-db
 cp /etc/default/sonic-db/database_config.json /var/run/redis/sonic-db/
 
 SYSTEM_MAC_ADDRESS=$(ip link show eth0 | grep ether | awk '{print $2}')
-sonic-cfggen -t /usr/share/sonic/templates/init_cfg.json.j2 -a "{\"system_mac\": \"$SYSTEM_MAC_ADDRESS\"}" > /etc/sonic/init_cfg.json
-
+if [ -f /etc/sonic/init_cfg.json ]; then
+    sonic-cfggen -j /etc/sonic/init_cfg.json -a "{\"system_mac\": \"$SYSTEM_MAC_ADDRESS\"}" > /tmp/init_cfg.json
+    mv /tmp/init_cfg.json  /etc/sonic/init_cfg.json
+else
+   sonic-cfggen -t /usr/share/sonic/templates/init_cfg.json.j2 -a "{\"system_mac\": \"$SYSTEM_MAC_ADDRESS\"}" > /etc/sonic/init_cfg.json
```
Comment (qiluo-msft) [inline platform/vs/docker-sonic-vs/start.sh:40]: >    [](http://example.com/codeflow?start=0&length=3)

use 4 blanks indentation.

File: platform/vs/docker-sonic-vs/Dockerfile.j2 (hunk 4 above)
Comment (qiluo-msft) [reply platform/vs/docker-sonic-vs/Dockerfile.j2:171]: This file is not existing during build.

File: rules/config [hunk 6]
```diff
@@ -187,6 +187,9 @@ SONIC_ENABLE_IMAGE_SIGNATURE ?= n
 # The absolute path should be provided.
 SONIC_ENABLE_SECUREBOOT_SIGNATURE ?= n
 
+# ENABLE_YANG_CONFIG_VALIDATION - enable YANG validation of updates to ConfigDB
+ENABLE_YANG_CONFIG_VALIDATION = n
```
Comment (qiluo-msft) [inline rules/config:191]: >ENABLE_YANG_CONFIG_VALIDATION [](http://example.com/codeflow?start=0&length=29)

Similar to synchronous_mode, do we really need a build time config?

**Comment by qiluo-msft** [comment]
Please resolve conflict

**Comment by isabelmsft** [comment]
This PR functionality replaced with new, merged PRs
https://github.com/sonic-net/sonic-buildimage/pull/12504
https://github.com/sonic-net/sonic-buildimage/pull/12497

**Pull request closed without merging**